### PR TITLE
Fix Broken MongoEngine Tests (All commits reference the breaking change)

### PR DESCRIPTION
**Pull request by iainb**

### CODE REVIEW

File: mongoengine/queryset.py [hunk 1]
```diff
@@ -511,6 +511,7 @@ def _cursor(self):
                 self._cursor_obj.sort(self._ordering)
             elif self._document._meta['ordering']:
                 self.order_by(*self._document._meta['ordering'])
+                self._cursor_obj.sort(self._ordering)
```
Comment (tjphopkins) [inline mongoengine/queryset.py]:
```
if self._document._meta['ordering']:
    self.order_by(*self._document._meta['ordering'])
self._cursor_obj.sort(self._ordering)
```

File: mongoengine/queryset.py [hunk 2]
```diff
@@ -511,6 +511,7 @@ def _cursor(self):
                 self._cursor_obj.sort(self._ordering)
             elif self._document._meta['ordering']:
                 self.order_by(*self._document._meta['ordering'])
```
Comment (tjphopkins) [inline mongoengine/queryset.py]: does order_by modify self._ordering?

Comment (iainb) [inline mongoengine/queryset.py]: Yes. I would recommend you look at the commits individually, they each reference a commit that caused the test to fail. You would be able to see that in the breaking commit.

File: mongoengine/queryset.py (hunk 1 above)
Comment (iainb) [inline mongoengine/queryset.py]: Wouldn't that apply the sort to the cursor regardless of `self._ordering` or `self._document._meta['ordering']` being true?

**Comment by tjphopkins** [comment]
🐳

**Pull request merged**